vnext/Microsoft.ReactNative/Fabric/Composition/ParagraphComponentView.cpp

```diff
@@ -299,6 +466,288 @@ void ParagraphComponentView::DrawText() noexcept {
   }
 }
 
+void ParagraphComponentView::ClearSelection() noexcept {
+  bool hadSelection = (m_selectionStart >= 0 && m_selectionEnd > m_selectionStart);
+  m_selectionStart = -1;
+  m_selectionEnd = -1;
+  m_isSelecting = false;
+  if (hadSelection) {
+    // Clears selection highlight
+    DrawText();
+  }
+}
+
+void ParagraphComponentView::OnPointerPressed(
+    const winrt::Microsoft::ReactNative::Composition::Input::PointerRoutedEventArgs &args) noexcept {
+  // Only handle selection if text is selectable
+  const auto &props = paragraphProps();
+  if (!props.isSelectable) {
+    Super::OnPointerPressed(args);
+    return;
+  }
+
+  auto pp = args.GetCurrentPoint(-1);
+
+  // Ignores right-click
+  if (pp.Properties().PointerUpdateKind() ==
+      winrt::Microsoft::ReactNative::Composition::Input::PointerUpdateKind::RightButtonPressed) {
+    args.Handled(true);
+    return;
+  }
+
+  auto position = pp.Position();
+
+  facebook::react::Point localPt{
+      position.X - m_layoutMetrics.frame.origin.x, position.Y - m_layoutMetrics.frame.origin.y};
+
+  int32_t charPosition = getTextPositionAtPoint(localPt);
+
+  if (charPosition >= 0) {
+    // Clears any previous text selection in another component
+    ClearCurrentTextSelection();
+
+    // Check for double-click
+    auto now = std::chrono::steady_clock::now();
+    auto timeSinceLastClick = std::chrono::duration_cast<std::chrono::milliseconds>(now - m_lastClickTime);
+    bool isDoubleClick = (timeSinceLastClick.count() < 500) && (m_lastClickPosition >= 0) &&
```

Comment:
This shouldn't be hard coded to 500ms.  There is a system setting for how long a double click is.